.github/actions/microrepo/version-non-npm/action.yaml

```diff
@@ -0,0 +1,44 @@
+name: Version and Release Non NPM Project
+
+inputs:
+  github-token:
+    type: string
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    # Even if we already have the code checked out, we need to check it out again with the PAT
+    - name: checkout_with_automation_pat
+      uses: actions/checkout@v3
+      with:
+        persist-credentials: true
+        fetch-depth: 0
+        token: ${{ inputs.github-token }}
+    # curl the auto binary and instead of installing it with npm to avoid installing node
+    # TODO: use a docker image with auto pre-installed
+    - name: download_auto
+      shell: bash
+      run: |
+        curl -vkL -o - https://github.com/intuit/auto/releases/download/v10.28.0/auto-linux.gz | gunzip > ~/auto
+        chmod +x ~/auto
+    # Instead of calling shipit, manually call each command
+    # This lets us control the tag names and avoids the plugins flow.
+    # This also lets us avoid having to commit an .autorc file into every repo
+    - name: run_auto_versioning_and_release
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.github-token }}
+      run: |
+        VERSION=`~/auto version`
```

Comment:
Yes by default it will patch, logic defined here https://intuit.github.io/auto/docs/configuration/autorc#label-customization. Currently we use auto's default logic and we would have to build a way to add configs that do not require any configs live in each repo. I also am not sure I agree with `minor` as a default. I could however see letting teams set this on a per repo bases.

Reply:
I think regardless, we'll end up with teams never deviating from the the default.  and to that end 1.0.287 and 1.287.0 aren't actually different.